File: cpp/tests/CombinationsTest.cpp
#include "Combinations.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(Combinations, Solution77_1) {
  auto S77_1 = Solution77_1();

  int const N1 = 4, K1 = 2;
  std::vector<std::vector<int>> const Expected1{{1, 2}, {1, 3}, {1, 4},
                                                {2, 3}, {2, 4}, {3, 4}};
  EXPECT_EQ(Expected1, S77_1.combine(N1, K1));

  int const N2 = 1, K2 = 1;
  std::vector<std::vector<int>> const Expected2{{1}};
  EXPECT_EQ(Expected2, S77_1.combine(N2, K2));
}

File: java/src/test/java/io/binac/leetcode/CombinationsTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.Collections;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class CombinationsTest {
    private static final Combinations.Solution1 SOLUTION1 = new Combinations.Solution1();

    private final int N1 = 4, K1 = 2;
    private final List<List<Integer>> EXPECTED1 = Arrays.asList(
            Arrays.asList(1, 2),
            Arrays.asList(1, 3),
            Arrays.asList(1, 4),
            Arrays.asList(2, 3),
            Arrays.asList(2, 4),
            Arrays.asList(3, 4)
    );

    private final int N2 = 1, K2 = 1;
    private final List<List<Integer>> EXPECTED2 = Collections.singletonList(Collections.singletonList(1));

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.combine(N1, K1));
        assertEquals(EXPECTED2, SOLUTION1.combine(N2, K2));
    }
}